core/types/receipt.go

```diff
@@ -508,37 +453,33 @@ func (r Receipts) deriveFieldsOrig(config *params.ChainConfig, hash common.Hash,
 
 // Quorum
 
-// storedMPSReceiptRLPWithRevertReason is the storage encoding of a receipt which contains
-// receipts per PSI, including Revert Reason
-type storedMPSReceiptRLPWithRevertReason struct {
-	PostStateOrStatus []byte
-	CumulativeGasUsed uint64
-	Logs              []*LogForStorage
-	PSReceipts        []storedPSIToReceiptMapEntryWithRevertReason
-	RevertReason      []byte
-}
-
-type storedPSIToReceiptMapEntryWithRevertReason struct {
-	Key   PrivateStateIdentifier
-	Value storedReceiptRLPWithRevertReason
+func (r *Receipt) SetReceiptExtraDataFromStorage(data *QuorumReceiptExtraData) {
+	if data == nil {
+		return
```

Comment:
no need to reset the fields?

Reply:
renamed to `FillReceiptExtraDataFromStorage`